scipy/signal/signaltools.py

```diff
@@ -463,6 +460,9 @@ def convolve(in1, in2, mode='full'):
     volume = asarray(in1)
     kernel = asarray(in2)
 
+    if not _is_valid_mode_shapes(mode, volume.shape, kernel.shape):
+        volume, kernel = kernel, volume
```

Comment:
I suppose we could, but as I pointed out earlier, in the general case, having to check whether the dimensions "violate" regardless of order will slow things down IMHO (you have to check that dimensions are violated in both directions).